configure

```diff
@@ -130,7 +136,13 @@ mac_64=false
 ismac64() {
 	$mac_64
 }
+solaris_detected=false
+issolaris() {
+	$solaris_detected
+}
 
+check_compile "whether fine to define _GNU_SOURCE" "-D_GNU_SOURCE" \
+'#include <unistd.h>\nint main() { return 0; }'
```

Comment:
does using _GNU_SOURCE cause build failure on solaris ?

Reply:
No, this is just to add `-D_GNU_SOURCE` to later check_compile calls.

Since the actual sources is compiled with `-D_GNU_SOURCE`, the test compiles should also define this; same reason for `C99_CFLAGS`.

Reply:
If you consider this `check_compile` is unnecessary, if can be replaced with `OUR_CPPFLAGS="$OUR_CPPFLAGS -D_GNU_SOURCE`

Reply:
yeah, the OUR_CPPFLAGS way is preferable